build/functions.j2

```diff
@@ -207,10 +208,44 @@ function {{ "scan_" ~ function if scan else function }}() {
       continue
     fi
 
+    ## Only rerun security tools if (1) it is the first time, or (2) we are able to detect a change
+    dirs_to_hash=("${dir}")
+  {%- if monitor is defined %}
+    {%- for env_var in monitor["env_vars"] if monitor["env_vars"] %}
+    if [[ -n {{ '${' }}{{ env_var }}} ]]; then
+      dirs_to_hash+=("{{ '${' }}{{ env_var }}}")
+    fi
+    {%- endfor %}
+  {%- endif %}
+    hash="$(sha256sum <(find ${dirs_to_hash[@]} -type f -exec sha256sum {} \; | sort) | awk '{print $1}')"
```

Comment:
I'm curious if we're just looking for changes to files, why use sha256 over md5? Is the hashing speed negligible in this use case?

Reply:
Yeah, two reasons - some environments cannot use anything that uses md5 regardless of why, and the speed difference wasn't enough to matter.  We should just be hashing tf files and provider directories.  Even hashing the easy_infra repo including the large `checkov` output didn't take _that_ long.